calculator_test.go

```diff
@@ -2,23 +2,167 @@ package calculator_test
 
 import (
 	"calculator"
+	"math"
+	"math/rand"
 	"testing"
 )
 
+type testCase struct {
+	a, b        float64
+	want        float64
+	comment     string
+	err         error
+	errExpected bool
+}
+
+var testCasesAdd = []testCase {
```

Comment:
Again, we would probably define the slice of test cases inside the specific test that uses them. This is an application of the general rule in Go that we want to define things in the smallest possible scope, to avoid accidental data leakage!